File: utils/format-name.ts
export function formatName(hyphenatedLowerCaseName: string): string {
    function sentenceCase(sentence: string | string[]) {
      const words = Array.isArray(sentence) ? sentence : sentence.split(" ");
      const res: string[] = [];
      for (let i = 0; i < words.length; i++) {
        const word = words[i];
        res.push(word[0].toUpperCase() + word.substring(1));
      }
      return res.join(" ");
    }
    return sentenceCase(hyphenatedLowerCaseName.split("-"));
  }

File: utils/formatname.go
package utils

import "strings"

func sentenceCase(sentence string) string {
	words := strings.Split(sentence, " ")
	res := make([]string, len(words))
	for i, word := range words {
		res[i] = strings.Title(word)
	}
	return strings.Join(res, " ")
}

func FormatName(hyphenatedLowerCaseName string) string {
	return sentenceCase(strings.Join(strings.Split(hyphenatedLowerCaseName, "-"), " "))
}
